Yoast/javascript
# Add prototype assessment for ranking intentions

## Summary


This PR can be summarized in the following changelog entry:

*  [yoastseo] Adds a prototype assessment to recognise ranking intentions in English. 
*  [Yoast SEO Premium] Adds a prototype assessment to recognise ranking intentions in English based on the ratio between singular and plural keyphrase occurrences in the text.


## Relevant technical choices:

* The assessment does not currently work on cornerstone articles, taxonomy articles and for related keyphrases.

## Test instructions

This PR can be tested by following these steps:

* Set your site to English;
* Test in Premium;
* Paste a text of at least 150 words;
* Set a singular keyphrase, e.g., `fluffy cat`;
* Add a number of occurrences of your singular keyphrase to the text;
* Add the same number of occurrences of the plural form of your keyphrase, i.e., `fluffy cats`;
* The ranking intention assessment should turn orange and read: `Your text does not reflect any particular ranking intention.  If your keywords is singular, use more singular occurrences; if your keyphrase is plural, use more plural occurrences!`;
* Change some of the singular occurrences to plural until the ranking intention bullet turns red and reads: `Your text does not reflect your ranking intention. Change your keyphrase occurrences!`;
* Change some of the plural occurrences to singular until the ranking intention bullet turns green and reads: `Your text reflects your ranking intention. Good job!`;
* Check that the same holds for a plural keyphrase;

* Check that the site does not crash and that no bullet is there for the assessment when you are in Free.
* Check that the assessment isn't shown when the site is set to another language than English.

Extra internal test instructions:
* Run yarn test and make sure all tests pass

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-544
